### ISSUE TITLE
[Bug]: Exception: Unknown error

### ISSUE THREAD

**Issue by Liu-Angelo** [acheong08/ChatGPT#719]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

        ChatGPT - A command-line interface to OpenAI's ChatGPT (https://chat.openai.com/chat)
        Repo: github.com/acheong08/ChatGPT
        
Logging in...
Logged in

Type '!help' to show a full list of commands
Press enter twice to submit your question.

You:
写一个霍夫曼树

ChatGPT:
error: Unknown error
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 364, in <module>
    asyncio.run(main())
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 353, in main
    async for line in chatbot.ask(prompt=prompt):
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 169, in ask
    raise Exception("Unknown error")
Exception: Unknown error

### Steps to reproduce the problem

1. Go to .... 
2. Press ....
3. ...

### What should have happened?

error: Unknown error
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 364, in <module>
    asyncio.run(main())
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 353, in main
    async for line in chatbot.ask(prompt=prompt):
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 169, in ask
    raise Exception("Unknown error")
Exception: Unknown error

### Version where the problem happens

Name: revChatGPT
Version: 2.2.0
Summary: ChatGPT is a reverse engineering of OpenAI's ChatGPT API
Home-page: https://github.com/acheong08/ChatGPT
Author: Antonio Cheong
Author-email: acheong@student.dalat.org
License: GNU General Public License v2.0
Location: /usr/local/lib/python3.10/site-packages
Requires: asyncio, httpx, OpenAIAuth, requests, tiktoken

### Command Line Arguments

```Shell
python3 -m revChatGPT.V2 -e ****.com -p *****
```

### Console logs

```Shell
error: Unknown error
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 364, in <module>
    asyncio.run(main())
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 353, in main
    async for line in chatbot.ask(prompt=prompt):
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 169, in ask
    raise Exception("Unknown error")
Exception: Unknown error
```

**Comment by reveching** [comment]
--insecure-auth if you're in China

**Comment by Liu-Angelo** [comment]
> ```shell
> python3 -m revChatGPT.V2 -e ****.com -p *****
> ```

python3 -m revChatGPT.V2 -e ****.com -p *****  --insecure-auth

Logging in...
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 364, in <module>
    asyncio.run(main())
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 304, in main
    chatbot = Chatbot(
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 135, in __init__
    self.login(email, password, proxy, insecure, session_token)
  File "/usr/local/lib/python3.10/site-packages/revChatGPT/V2.py", line 224, in login
    self.api_key = auth_request.json()["accessToken"]
KeyError: 'accessToken'

**Comment by hucs2542** [comment]
I encountered the same problem as him. I set up the connection via Socket5, and GPT works fine on the browser.

**Comment by mikebilly** [comment]
same problem here

**Comment by getxd-dev** [comment]
either like cloudflare bypass server got taken down or OpenAI patched the bypass.

**Comment by fengqingthu** [comment]
Same problem seen here.

**Comment by bywjge** [comment]
Same problem here.

**Comment by readpan** [comment]
Same

**Comment by go-xmyang** [comment]
same problem here

**Comment by sqy941013** [comment]
me too, I ran flask server on azure US region.

**Comment by Pengchengistaken** [comment]
`ChatGPT:
error: Unknown error
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/ubuntu/revChatGPT/lib/python3.8/site-packages/revChatGPT/V2.py", line 364, in <module>
    asyncio.run(main())
  File "/usr/lib/python3.8/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/ubuntu/revChatGPT/lib/python3.8/site-packages/revChatGPT/V2.py", line 353, in main
    async for line in chatbot.ask(prompt=prompt):
  File "/home/ubuntu/revChatGPT/lib/python3.8/site-packages/revChatGPT/V2.py", line 169, in ask
    raise Exception("Unknown error")
Exception: Unknown error
(revChatGPT) ubuntu@instance-20220820-1045:~$ pip3 list *GPT*
Package                 Version
----------------------- ----------
anyio                   3.6.2
async-generator         1.10
asyncio                 3.4.3
attrs                   22.2.0
blobfile                2.0.1
certifi                 2022.12.7
charset-normalizer      3.0.1
exceptiongroup          1.1.0
filelock                3.9.0
h11                     0.14.0
httpcore                0.16.3
httpx                   0.23.3
idna                    3.4
lxml                    4.9.2
OpenAIAuth              0.2.0
outcome                 1.2.0
pip                     20.0.2
pkg-resources           0.0.0
pycryptodomex           3.17
PySocks                 1.7.1
regex                   2022.10.31
requests                2.28.2
revChatGPT              2.2.0
rfc3986                 1.5.0
selenium                4.8.0
setuptools              44.0.0
sniffio                 1.3.0
socksio                 1.0.0
sortedcontainers        2.4.0
tiktoken                0.2.0
tls-client              0.1.8
trio                    0.22.0
trio-websocket          0.9.2
undetected-chromedriver 3.4.6
urllib3                 1.26.14
websockets              10.4
wsproto                 1.2.0`

**Comment by zhangjiany2929** [comment]
https://chat.duti.tech/ 502 bad gateway

**Comment by wangcailin** [comment]
> https://chat.duti.tech/ 502 bad gateway

That's what happened to me！

**Comment by kevindany** [comment]
Same issue, even I run this command without proxy(on server outside of China). So probably the command running chatgpt was detected by openai?

**Comment by Lxk-Kallen** [comment]
same

**Comment by acheong08** [comment]
Sorry guys. Server got overloaded due to memory leak in `--insecure-auth` proxy. No more insecure auth allowed

**Comment by acheong08** [comment]
Fixed now

**Comment by LotusMoves** [comment]
im confused can we not run local does it have to go though a proxy?

**Comment by acheong08** [comment]
> im confused can we not run local does it have to go though a proxy?

It is to prevent OpenAI from finding the cloudflare bypass

**Comment by LotusMoves** [comment]
i take it the local browser method no longer works?

**Comment by acheong08** [comment]
Still works somewhat. Just not maintained

**Comment by Pengchengistaken** [comment]
> Fixed now

love you

**Comment by LotusMoves** [comment]
any reason to not maintain it? i feel like like that would be the best way right? everyone self host at that point

**Comment by acheong08** [comment]
> any reason to not maintain it? i feel like like that would be the best way right? everyone self host at that point

It is very difficult to deal with browsers as a developer as everyone has different versions which causes a ton of issues. 

Another reason is that my IP address is banned by Cloudflare and I don't have a reliable proxy. I must use the Cloudflare bypass to access ChatGPT.

**Comment by zjrwtx** [comment]
> Fixed now
![image](https://user-images.githubusercontent.com/86822589/218636644-d8157027-80fa-4e80-aa84-0b7e3b70a8a1.png)
![image](https://user-images.githubusercontent.com/86822589/218636666-0d62f6c9-eb42-4048-904e-23370806d229.png)
still not work for me

**Comment by zjrwtx** [comment]
it is my error?

> > Fixed now
> > ![image](https://user-images.githubusercontent.com/86822589/218636644-d8157027-80fa-4e80-aa84-0b7e3b70a8a1.png)
> > ![image](https://user-images.githubusercontent.com/86822589/218636666-0d62f6c9-eb42-4048-904e-23370806d229.png)
> > still not work for me

**Comment by LotusMoves** [comment]
> > any reason to not maintain it? i feel like like that would be the best way right? everyone self host at that point
> 
> It is very difficult to deal with browsers as a developer as everyone has different versions which causes a ton of issues.
> 
> Another reason is that my IP address is banned by Cloudflare and I don't have a reliable proxy. I must use the Cloudflare bypass to access ChatGPT.

this kinda makes sense but you can force everyone to whatever browser and version you want. as for being banned by Cloudflare (nice lol) idk where you are obv but I do know if your in the states and you rolled your own firewall you can force the isp to get you a new IP buy running a release renew on that port. other than that I would be down to try and help with the local browser implementation but again idk what the main differences are yet I haven't dug into your code

**Comment by acheong08** [comment]
> this kinda makes sense but you can force everyone to whatever browser and version you want.

Not possible

**Comment by acheong08** [comment]
I have a static IP and my entire ISP IP range got blocked

**Comment by LotusMoves** [comment]
> > this kinda makes sense but you can force everyone to whatever browser and version you want.
> 
> Not possible

why? do you have a reason for this?

and that sucks

**Comment by acheong08** [comment]
I can technically bring the browserless code in but OpenAI will disable the method in about an hour. That is my main concern

**Comment by LotusMoves** [comment]
> I can technically bring the browserless code in but OpenAI will disable the method in about an hour. That is my main concern

no keep that as a fail over backup 
I was just wondering about the browser method I don't mind haveing to tell the browser im a human now and then

**Comment by acheong08** [comment]
> why? do you have a reason for this?

Technical limitations. I can't detect every platform and install a proper browser for them

**Comment by LotusMoves** [comment]
> > why? do you have a reason for this?
> 
> Technical limitations. I can't detect every platform and install a proper browser for them

no I mean make it a requirement tell users runs on chrome version whatever

**Comment by acheong08** [comment]
> no I mean make it a requirement tell users runs on chrome version whatever

It already does that when the wrong version is installed. Many users can't figure out how to install chrome properly

**Comment by acheong08** [comment]
> no keep that as a fail over backup

The point is to keep it simple for most users. They can choose to host their own bypass server or use the browser version if they want

**Comment by acheong08** [comment]
But that is more complicated and difficult for users to do

**Comment by acheong08** [comment]
While deploying to a VPS, I personally spent hours on getting the browser version working

**Comment by LotusMoves** [comment]
> But that is more complicated and difficult for users to do

again the easy way would be the proxy and the self host requires some work I mean we are on github lol your not releasing this as a all in one downloadable app lol

**Comment by acheong08** [comment]
> > But that is more complicated and difficult for users to do
> 
> again the easy way would be the proxy and the self host requires some work I mean we are on github lol your not releasing this as a all in one downloadable app lol

You have no idea how many issues are duplicates of just Chrome being of the wrong version. It's very distracting.

The proxy code itself is actually open source https://github.com/RunawayVPN/PROX with a missing config

**Comment by LotusMoves** [comment]
> > > But that is more complicated and difficult for users to do
> > 
> > 
> > again the easy way would be the proxy and the self host requires some work I mean we are on github lol your not releasing this as a all in one downloadable app lol
> 
> You have no idea how many issues are duplicates of just Chrome being of the wrong version. It's very distracting.
> 
> The proxy code itself is actually open source https://github.com/RunawayVPN/PROX with a missing `config.json`

trust me I know lol but you can't help them the console literally says when you went up a chrome version i read the error and fixed it if they cant read you cant help

**Comment by Pengchengistaken** [comment]
> 

aggred, you are not releasing a commercial product.
